# 2. Создайте собственный класс-исключение, обрабатывающий ситуацию деления на нуль.
# Проверьте его работу на данных, вводимых пользователем.
# При вводе пользователем нуля в качестве делителя программа должна корректно обработать эту ситуацию и не завершиться с ошибкой.


class OwnError(ZeroDivisionError):
    def __init__(self, txt):
        self.txt = txt


inp_data = int(input("Введите положительное число: "))

try:
    if inp_data == 0:
        raise OwnError("Вы пытаетесь поделить на 0")

except OwnError as err:
    print(err)
else:
    res = 100 / inp_data
    print(f"Все хорошо. Ваш результат: {res}")